prime-router/src/main/kotlin/fhirengine/engine/FHIRReceiverFilter.kt

```diff
@@ -28,24 +57,397 @@ class FHIRReceiverFilter(
     azureEventService: AzureEventService = AzureEventServiceImpl(),
     reportService: ReportService = ReportService(),
 ) : FHIREngine(metadata, settings, db, blob, azureEventService, reportService) {
-    /**
-     * Adds logs for reports that pass through various methods in the FHIRRouter
-     */
-    // private var actionLogger: ActionLogger? = null
+    override val finishedField: Field<OffsetDateTime> = Tables.TASK.RECEIVER_FILTERED_AT
+
+    override val engineType: String = "ReceiverFilter"
 
     /**
-     * Process a [message] off of the raw-elr azure queue, convert it into FHIR, and store for next step.
+     * Accepts a [message] in internal FHIR format
+     *
+     * [message] is the incoming message to be evaluated for valid receivers and routed to the appropriate queues
      * [actionHistory] and [actionLogger] ensure all activities are logged.
      */
     override fun <T : QueueMessage> doWork(
         message: T,
         actionLogger: ActionLogger,
         actionHistory: ActionHistory,
-    ): List<FHIREngineRunResult> {
-        // TODO: implement receiver filter (see: https://github.com/CDCgov/prime-reportstream/issues/13692)
-        return emptyList()
+    ): List<FHIREngineRunResult> =
+        when (message) {
+            is FhirReceiverFilterQueueMessage -> {
+                check(message.topic.isUniversalPipeline) {
+                    "Unexpected topic $message.topic in the Universal Pipeline routing step."
+                }
+                fhirEngineRunResults(message, actionLogger, actionHistory)
+            }
+
+            else -> {
+                throw RuntimeException(
+                    "Message was not a FhirReceiverFilter and cannot be processed: $message"
+                )
+            }
+        }
+
+    class ReceiverItemFilteredActionLogDetail(
+        val filter: String,
+        @JsonProperty
+        val filterType: ReportStreamFilterType,
+        @JsonProperty
+        val receiverOrg: String,
+        @JsonProperty
+        val receiverName: String,
+        val index: Int = 1,
+    ) : ActionLogDetail {
+        override val scope: ActionLogScope = ActionLogScope.item
+        override val message: String =
+            "Item was not routed to $receiverName because it did not pass the $filterType. Item failed on: $filter"
+
+        override val errorCode: ErrorCode = ErrorCode.UNKNOWN
```

Comment:
This is a leftover of ActionLogDetail having an `errorCode` member. This errorCode is not appropriate since ActionLogDetail is not always an error.

In other places in the codebase we did the same (UNKNOWN ErrorCode for ActionLogDetail that isnt an error) but I agree this is not a good pattern.

This probably warrants a tech debt ticket to address throughout codebase?